packages/superset-ui-legacy-preset-chart-nvd3/src/NVD3Vis.js

```diff
@@ -607,16 +609,54 @@ function nvd3Vis(element, props) {
       xTicks.selectAll('text').attr('dx', -6.5);
     }
 
-    // Apply y-axis bounds
-    if (chart.yDomain && Array.isArray(yAxisBounds) && yAxisBounds.length === 2) {
-      const [min, max] = yAxisBounds;
-      const hasCustomMin = isDefined(min) && !Number.isNaN(min);
-      const hasCustomMax = isDefined(max) && !Number.isNaN(max);
-      if (hasCustomMin && hasCustomMax) {
-        chart.yDomain([min, max]);
-        chart.clipEdge(true);
+    const applyYAxisBounds = () => {
+      if (chart.yDomain && Array.isArray(yAxisBounds) && yAxisBounds.length === 2) {
+        const [customMin, customMax] = yAxisBounds;
+        const hasCustomMin = isDefined(customMin) && !Number.isNaN(customMin);
+        const hasCustomMax = isDefined(customMax) && !Number.isNaN(customMax);
+
+        if ((hasCustomMin || hasCustomMax) && vizType === 'area' && chart.style() === 'expand') {
+          // Because there are custom bounds, we need to override them back to 0%-100% since this
+          // is an expanded area chart
+          chart.yDomain([0, 1]);
+        } else if (
+          (hasCustomMin || hasCustomMax) &&
+          vizType === 'area' &&
+          chart.style() === 'stream'
+        ) {
+          // Because there are custom bounds, we need to override them back to the domain of the
+          // chart since this is a stream area chart
+          chart.yDomain(computeStackedYDomain(data));
+        } else if (hasCustomMin && hasCustomMax) {
+          // Override the y domain if there's both a custom min and max
+          chart.yDomain([customMin, customMax]);
+          chart.clipEdge(true);
+        } else if (hasCustomMin || hasCustomMax) {
+          // Only one of the bounds has been set, so we need to manually calculate the other one
+          let [trueMin, trueMax] = [0, 1];
+
+          // These viz types can be stacked
+          // They correspond to the nvd3 stackedAreaChart and multiBarChart
+          if (vizType === 'area' || (isVizTypes(['bar', 'dist_bar']) && chart.stacked())) {
+            // This is a stacked area chart or a stacked bar chart
+            [trueMin, trueMax] = computeStackedYDomain(data);
+          } else {
+            [trueMin, trueMax] = computeYDomain(data);
+          }
+
+          const min = hasCustomMin ? customMin : trueMin;
+          const max = hasCustomMax ? customMax : trueMax;
+          chart.yDomain([min, max]);
+          chart.clipEdge(true);
+        }
       }
-    }
+    };
+    applyYAxisBounds();
+
+    // Also reapply on each state change to account for enabled/disabled series
+    chart.dispatch.on('stateChange.applyYAxisBounds', () => {
```

Comment:
can we remove the extra arrow func here? 

`chart.dispatch.on('stateChange.applyYAxisBounds', applyYAxisBounds)`

Reply:
good catch, done!